pyscitt/pyscitt/cli/sign_claims.py

```diff
@@ -110,16 +122,81 @@ def sign_claims(
     feed: Optional[str],
     registration_info_args: List[RegistrationInfoArgument],
     x5c_path: Optional[str],
+    akv_configuration_path: Optional[Path],
 ):
-    signer = create_signer_from_arguments(
-        key_path, did_doc_path, kid, issuer, algorithm, x5c_path
-    )
-    claims = claims_path.read_bytes()
-    registration_info = {arg.name: arg.value() for arg in registration_info_args}
+    # If a Key Vault configuration is provided, we sign with AKV
+    if akv_configuration_path:
+        # Parse the AKV configuration file
+        akv_sign_configuration_dict = json.loads(akv_configuration_path.read_text())
 
-    signed_claims = crypto.sign_claimset(
-        signer, claims, content_type, feed, registration_info
-    )
+        vault_name = akv_sign_configuration_dict["keyVaultName"]
+        vault_url = f"https://{vault_name}.vault.azure.net"
+        identity_certificate_name = akv_sign_configuration_dict["certificateName"]
+        identity_certificate_version = akv_sign_configuration_dict["certificateVersion"]
+
+        # We need to use SecretClient to get the full certificate chain from Key Vault,
+        # as we need to add it to the COSE payload in the x5c header.
+        # The CertificateClient only fetches the leaf certificate.
+        secret_client = SecretClient(
+            vault_url=vault_url, credential=DefaultAzureCredential()
+        )
+
+        certificate_secret = secret_client.get_secret(
+            name=identity_certificate_name,
+            version=identity_certificate_version,
+        )
+
+        # We need to check that the certificate is a PEM certificate
+        if certificate_secret.properties.content_type != "application/x-pem-file":
+            raise ValueError(
+                "Certificate content type is not supported. Please use PEM certificates instead."
+            )
+
+        assert (
+            isinstance(certificate_secret.value, str) and certificate_secret.value
+        ), "Invalid identity public certificate available for this identity"
+
+        def _remove_substring(s: str, start_delim: str, end_delim: str) -> str:
+            """Utility method to remove a substring from a string
+            given the start and end delimiters"""
+
+            start = s.find(start_delim)
+            end = s.find(end_delim)
+            if start != -1 and end != -1:
+                return s[:start] + s[end + len(end_delim) :]
+            else:
+                return s
+
+        # Remove the private key part from the certificate secret
+        public_certs_string = _remove_substring(
```

Comment:
This holds for the cases where 2 self signed cert copies need to be in the chain as well which would not otherwise be possible with this approach

Reply:
That was my intention too, but unfortunately it's not possible to download the entire certificate chain from AKV without fetching the whole secret. I tested with a `CertificateClient` and getting the certificate only returns the leaf cert, which is not enough to include in the x5c header (it will get rejected at the time the claim is submitted to SCITT). 

As far as I could see in other SO posts and other resource ([example](https://learn.microsoft.com/en-us/answers/questions/441132/retrieving-certificate-with-complete-chain-from-az)), downloading the certificate chain can only be done this way, and unfortunately it has the downside of downloading the private key (assuming the private key can be exported, otherwise it will be missing). That is why I fell back to this solution (there is also a comment above detailing this a bit more).

Reply:
If it's preferred to ask users to provide the complete cert chain as a separate argument, it's fine and we can do that though. I just thought it was simpler to download it from AKV directly without requiring users to do the same on their end.

Reply:
I think we should not do it, this is similar to what Notation does as well - asks users for a chain explicitly. Another thing is that it forces users to have a policy to access secrets which would most likely be restricted anyway.

Reply:
Made the change in https://github.com/microsoft/scitt-ccf-ledger/pull/172/commits/0f8410977d0a6fa9ae987f4c6724da32aded5fd4